File: projects/developer-terminal-c-and-vb/Developer Term VB Beta r02/LED22.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Text
Imports System.Windows.Forms

Namespace LED
	Public Partial Class LED22
		Inherits UserControl
		Public Enum LEDColor
			RedOn
			RedOff
			YellowOn
			YellowOff
			OrangeOn
			OrangeOff
			GreenOn
			GreenOff
			BlueOn
			BlueOff
		End Enum

		Public Sub New()
			InitializeComponent()

			Me.BackgroundImage = imageList1.Images(CInt(Fix(LEDColor.RedOn)))

			setImage(CInt(Fix(LEDColor.RedOn)))
		End Sub

		Private Image_Renamed As Integer = CInt(Fix(LEDColor.RedOn))
		Public Property image() As Integer
			Get
				Return Image_Renamed
			End Get
			Set(ByVal value As Integer)
				Image_Renamed = value
				setImage(Image_Renamed)
			End Set
		End Property

		Private Sub setImage(ByVal num As Integer)
			If num <= CInt(Fix(LEDColor.BlueOff)) Then
				'MessageBox.Show("Set backgroung image to: " + num.ToString());
				Me.BackgroundImage = imageList1.Images(num)
			Else
				MessageBox.Show("Error: LED control, setImage - num > LEDColor.BlueOff.")
			End If
		End Sub

	End Class
End Namespace

File: projects/developer-terminal-c-and-vb/Developer Term Csharp Beta r02/LED22.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;

namespace LED
{
    public partial class LED22 : UserControl
    {
        public enum LEDColor { RedOn, RedOff, YellowOn, YellowOff, OrangeOn, OrangeOff, GreenOn, GreenOff, BlueOn, BlueOff }
        
        public LED22()
        {
            InitializeComponent();

            this.BackgroundImage = imageList1.Images[(int)LEDColor.RedOn];

            setImage((int)LEDColor.RedOn);
        }

        private int Image = (int)LEDColor.RedOn;
        public int image
        {
            get
            {
                return Image;
            }
            set
            {
                Image = value;
                setImage(Image);
            }
        }

        private void setImage(int num)
        {
            if (num <= (int)LEDColor.BlueOff)
            {
                //MessageBox.Show("Set backgroung image to: " + num.ToString());
                this.BackgroundImage = imageList1.Images[num];
            }
            else MessageBox.Show("Error: LED control, setImage - num > LEDColor.BlueOff.");
        }

    }
}
